src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -251,6 +250,18 @@ public object GetProperyValue(object obj, string propertyName)
 
             fields.Apply(field =>
             {
+                if (objectType is IReturnsAllFields)
```

Comment:
Yes, on second thought that was exactly what I realised too. So I went ahead and removed it.